tests/Language/Haskell/Stylish/Step/Imports/Tests.hs

```diff
@@ -930,3 +1009,248 @@ case38 = assertSnippet (step (Just 80) $ fromImportAlign File)
+          , groupPatterns = ["Project\\.Internal", "Project", "([^.]+)"]
+          , importAlign   = None
+          , longListAlign = Multiline
+          }
+
+--------------------------------------------------------------------------------
+case44a :: Assertion
+case44a =
+    assertSnippet (step (Just 80) options)
+    [ "import Project"
+    , "import Control.Monad"
+    , ""
+    , "import qualified Data.Acid as Acid"
+    , "import Project.Something"
+    , "import Data.Default.Class (Default (def))"
+    , ""
+    , "import qualified Herp.Derp.Internal.Types.Foobar as Internal (foo, bar)"
+    , "import ProJect.WrongCapitalization"
+    ]
+    [ "import Project"
+    , "import Project.Something"
+    , ""
+    , "import Control.Monad"
+    , "import qualified Data.Acid as Acid"
+    , "import Data.Default.Class (Default (def))"
+    , "import qualified Herp.Derp.Internal.Types.Foobar as Internal (bar, foo)"
+    , "import ProJect.WrongCapitalization"
+    ]
+  where options = defaultOptions
+          { groupImports  = True
+          , groupPatterns = ["Project"]
+          , importAlign   = None
+          }
+
+--------------------------------------------------------------------------------
+case44b :: Assertion
+case44b =
+    assertSnippet (step (Just 80) options)
+    [ "import Project"
+    , "import Control.Monad"
+    , ""
+    , "import qualified Data.Acid as Acid"
+    , "import Project.Something"
+    , "import Data.Default.Class (Default (def))"
+    , ""
+    , "import qualified Herp.Derp.Internal.Types.Foobar as Internal (foo, bar)"
+    , "import ProJect.WrongCapitalization"
+    ]
+    [ "import Project"
+    , "import Project.Something"
+    , ""
+    , "import qualified Data.Acid as Acid"
+    , ""
+    , "import Data.Default.Class (Default (def))"
+    , ""
+    , "import qualified Herp.Derp.Internal.Types.Foobar as Internal (bar, foo)"
+    , ""
+    , "import Control.Monad"
+    , ""
+    , "import ProJect.WrongCapitalization"
+    ]
+  where options = defaultOptions
+          { groupImports  = True
+          , groupPatterns = ["Project", "[^.]+\\.([^.]+)"]
+          , importAlign   = None
+          }
+
+
+--------------------------------------------------------------------------------
+case44c :: Assertion
+case44c =
+    assertSnippet (step (Just 80) options)
+    [ "import Project"
+    , "import Control.Monad"
+    , ""
+    , "import qualified Data.Acid as Acid"
+    , "import Project.Something"
+    , "import Data.Default.Class (Default (def))"
+    , ""
+    , "import qualified Herp.Derp.Internal.Types.Foobar as Internal (foo, bar)"
+    , "import ProJect.WrongCapitalization"
+    ]
+    [ "import Project"
+    , "import Project.Something"
+    , ""
+    , "import Control.Monad"
+    , "import qualified Data.Acid as Acid"
+    , "import Data.Default.Class (Default (def))"
+    , "import qualified Herp.Derp.Internal.Types.Foobar as Internal (bar, foo)"
+    , "import ProJect.WrongCapitalization"
```

Comment:
Wait, *now* we're case-sensitive? That seems inconsistent.

Reply:
Yeah, that's a bit odd. To make things consistent, I made all of the subgrouping logic case sensitive.